### DISCUSSION TITLE
Fragment ion intensities

### DISCUSSION THREAD

**Discussion by GolAGitHub** [vdemichev/DiaNN#425] [Q&A]
Hello,

I would like to extract fragment ion intensities following a search of DIA data. How can I map the ";"-separated values from the column "Fragment.Quant.Raw" in the report to the "FragmentType" / "FragmentCharge" / "FragmentSeriesNumber" specified in report-lib?

Thank you!

**Comment by vdemichev** [comment]
Hi, please add --report-lib-info to 'Additional options', the main report will then contain fragment annotation.

**Reply by GolAGitHub** [reply]
Thank you!

**Comment by GolAGitHub** [comment]
Hello there!

Following my recent inquiry, I was able to retrieve fragmentation annotations. However, looking at the information retrieved, I noticed that in a few instances, the annotations contained duplicated ions. Here is an example:

y8^2/441.740448;y8^1/882.4730835;y6^1/714.3831787;y7^1/785.4202881;y5^1/551.3198853;y3^1/393.2507324;y4^1/464.2878418;y9^2/498.2824707;b3^1/340.1872559;y7^1/393.2140503;y6^2/357.6954956;b6^1/661.3197021;

y7^1 appears twice. I simply seems that the second occurrence should be y7^2

Although the m/z values after the “\” are correct (714.3831787 for the first 393.2140503 for the second), this could potentially cause some issues for those (like me) relying exclusively on the identity of the ion, and should probably be corrected in the future.

I’ll be happy to provide additional information if needed.

Kind regards.
Guillaume

**Reply by vdemichev** [reply]
Hi Guillaume

How are the respective fragments annotated in the spectral library you are using?

Best,
Vadim

**Comment by GolAGitHub** [comment]
Hi Vadim,

The library was created by DIANN directly. I presume these are the spectral library files:
report-lib.tsv.speclib
report-lib.predicted.speclib

If these are the correct files, what application should I use to open them?

Thank you!
Guillaume

**Reply by vdemichev** [reply]
Any .speclib file can be converted using DIA-NN to .tsv, so the respective .tsv lib can be examined using R/Python
